src/ldp/http/metadata/BasicMetadataSerializer.ts

```diff
@@ -0,0 +1,26 @@
+import type { HttpResponse } from '../../../server/HttpResponse';
+import type { RepresentationMetadata } from '../../representation/RepresentationMetadata';
+import { MetadataSerializer } from './MetadataSerializer';
+import type { MetadataWriter } from './MetadataWriter';
+
+/**
+ * Basic serializer that calls all its {@link MetadataWriter}s sequentially.
+ */
+export class BasicMetadataSerializer extends MetadataSerializer {
```

Comment:
Yea I expected a comment about the names here. :D Probably should have written something about it in the PR.

What is happening here is exactly the same as when we parse metadata. We have 1 main handler that calls a bunch of parsers (writers here) that each do their thing on the input. So we want all of them to run.

I just realize that we had a similar discussion about this for the metadata parser. I think my argument there was that handlers sort of have a different contract so I felt they didn't fit here. (Also in the case of the metadata parsing some more had to be done before calling them so couldn't be just a composite handler).

Reply:
But here, the contract is the exact same IMHO. So the comment is not about the names, but about the contract being essentially the same.

Reply:
The same as what? It's not the same as AsyncHandler since we want each of our handlers to run, not just the first that can (since in this case we want both metadata writers to trigger, both the mapped one and the linkrel one).

MetadataSerializer is an AsyncHandler, but the things it calls (Writers) aren't.

Reply:
I am suggesting to make `MetadataWriter` also an `AsyncHandler`, given how close the contracts are:

```
  write: (response: HttpResponse, metadata: RepresentationMetadata) => Promise<void>;
```

versus

```
public async handle({ response: HttpResponse; metadata: RepresentationMetadata }): Promise<void> 
```

And yes, there's also `canHandle`, but see my comment about a base class 🙂

Reply:
They could even have a `canHandle` where they check if the metadata contains any triples relevant to them. No strong opinion there. Would need to make a new composite class that executes all handlers instead of just the first succeeding one though.

Reply:
> Would need to make a new composite class that executes all handlers instead of just the first succeeding one though.

Exactly. The existing one would be a `FirstCompositeHandler` or `AnyCompositeHandler` and the second an `AllCompositeHandler` or so. Makes a lot of sense to me.